rbi/core/hash.rbi

```diff
@@ -362,6 +362,22 @@ class Hash < Object
   sig {returns(T::Enumerator[[K, V]])}
   def delete_if(&blk); end
 
+  # Retrieves the value object corresponding to the each key objects repeatedly.
+  #
+  # ~~~ruby
+  # h = { foo: {bar: {baz: 1}}}
+  #
+  # h.dig(:foo, :bar, :baz)           #=> 1
+  # h.dig(:foo, :zot)                 #=> nil
+  # ~~~
+  sig do
+    params(
+      key: K
+    )
+    .returns(T.nilable(BasicObject))
```

Comment:
I'd expect this type to be too restrictive and most usages are thus likely to fail.
I don't think we can prescribe a type other that T.untyped for this method in practice without writing custom c++ for it.

Reply:
I changed the signature so it returns a `T.untyped`.

@elliottt let me know if it still produces errors in mono-repo (and I can't access the link you provided).

Reply:
There are now 11 errors in the mono-repo, and they look like they're related to the internal version of hidden.rbi. I'm going to hold off on merging this for now, but otherwise it looks good.